src/core/basetypes/MCData.cpp

```diff
@@ -664,23 +691,22 @@ Data * Data::decodedDataUsingEncoding(Encoding encoding)
                 if (length == 0) {
                     break;
                 }
-                decoded_length = uudecode(current_p, length);
+                decoded_length = uudecode(current_p, length, decoded_buf, sizeof(decoded_buf));
```

Comment:
Did you notice big memory usage is real use case for uudecode()?
It doesn't seem like an encoding format that's used a lot.
I would be fine with keeping the previous implementation of the decoding uuencoded parts.